.circleci/config.yml

```diff
@@ -17,101 +19,114 @@ workflows:
  
 jobs:
   validate-code:
-    working_directory: ~/code
-    docker:
-      - image: circleci/android:api-29-node
-    environment:
-      JVM_OPTS: -Xmx3200m
+    executor:
+      name: android/android-docker
+      resource-class: medium
+      tag: 2024.01.1
     steps:
       - checkout
 
-      - run:
-          name: Check Code Format
-          command: make format-check
+      - android/restore-gradle-cache
+
+      - android/restore-build-cache
+
+      # TODO: Re-enable these steps once formatting and linting issues are resolved
+      # - run:
+      #     name: Check code format
+      #     command: make checkformat
+      
+      # - run:
+      #     name: Run lint
+      #     command: make lint
+
+      - android/save-gradle-cache
+
+      - android/save-build-cache
 
   build-and-unit-test:
-    working_directory: ~/code
-    docker:
-      - image: circleci/android:api-29-node
-    environment:
-      JVM_OPTS: -Xmx3200m
+    executor:
+      name: android/android-docker
+      resource-class: medium
+      tag: 2024.01.1
     steps:
       - checkout
 
-      - run:
-          name: Javadoc
-          command: make ci-javadoc
-      - store_artifacts:
-          path: ci/javadoc/build/reports
+      - android/restore-gradle-cache
+
+      - android/restore-build-cache
 
       - run:
-          name: Build
-          command: make ci-build
+          name: Build Javadoc
+          command: |
+            make javadoc
```

Comment:
If it's a single line, you don't need to pipe it. Just use `command: make javadoc`